src/mock_parsec/tests.rs

```diff
@@ -204,6 +204,77 @@ fn consensus_mode_single() {
     assert_eq!(alice_blocks, bob_blocks);
 }
 
+#[test]
+fn reevaluate_previously_insufficient_votes() {
+    init_mock();
+
+    // Create 5 nodes (A, B, C, D, E). 3 of them vote to remove D, which is not enough to reach
+    // consensus. Then all vote to remove E which gets consensused and E is removed. The previous
+    // votes to remove D are now enough to reach consensus and so D should be removed too.
+
+    let genesis_group: BTreeSet<_> = (0..5).map(|n| PeerId(n)).collect();
+    let mut nodes: Vec<_> = genesis_group
+        .iter()
+        .map(|peer_id| {
+            Parsec::<Payload, _>::from_genesis(
+                Default::default(),
+                peer_id.clone(),
+                &genesis_group,
+                ConsensusMode::Single,
+            )
+        })
+        .collect();
+
+    // Drain the `Genesis` blocks.
+    gossip_all(&mut nodes);
+    for node in &mut nodes {
+        for _ in poll_all(node) {}
```

Comment:
why the `for _ in` instead of only `poll_all(node)`?

Reply:
`poll_all` returns an iterator. I needed to consume it.